lib/xclarity_client/connection/connection.rb

```diff
@@ -31,8 +34,8 @@ def initialize(configuration)
     #
     def do_get(uri = "", query = {})
       url_query = query.size > 0 ? "?" + query.map {|k, v| "#{k}=#{v}"}.join("&") : ""
-      @connection.get(uri + url_query)
-    rescue Faraday::Error::ConnectionFailed => e
+      Timeout.timeout(@timeout) { @connection.get(uri + url_query) }
+    rescue Faraday::Error::ConnectionFailed, Timeout::Error => e
```

Comment:
__Question__
Once that `Faraday::Error::ConnectionFailed` and `Timeout::Error` have different meanings, I think that if we create different log messages will be better to track errors. What do you think?